atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/Blacklist.java

```diff
@@ -108,8 +106,8 @@ private boolean isHostHealthy(CassandraClientPoolingContainer container) {
         }
     }
 
-    public Set<CassandraServer> filterBlacklistedHostsFrom(Collection<CassandraServer> potentialHosts) {
-        return Sets.difference(ImmutableSet.copyOf(potentialHosts), blacklist.keySet());
```

Comment:
![image](https://user-images.githubusercontent.com/54594/172282162-34829285-97d9-4071-8298-cfbe494f02ae.png)

Reply:
We should be able to make the argument an `ImmutableSet`, no? The only annoying one is in the use of `getRandomGoodHostForPredicate` - the `Sets.filter` should be an immutable result (since the input is immutable - at least, I assume that's how that works!) but I think we'd need to use another `copyOf` to get the right type. Alternatively, if you reverted that back to a stream and used `ImmutableSet.toImmutableSet()` we'd have the right types everywhere.